app/routes/views/room-detail/components/game-grid.tsx

```diff
@@ -42,6 +42,7 @@ export function GameGrid({ className, ...props }: GameGridProps) {
         <FlipCard
           key={card.id}
           onClick={() => handleCardClick(card.id)}
+             disabled={card.isMatched ||  card.id === firstSelectedCardId}
```

Comment:
this should be the only change needed for this PR as discussed in the github issue.